pravega/bookkeeper-operator
# Issue 46: Migration to helm v3, making cluster specs overridable and adding self-validating hooks

Signed-off-by: SrishT <Srishti.Thakkar@dell.com>

### Change log description
Updates the charts and the the helm commands to work with helm v3. It makes all the cluster specs overridable through Helm values. Also adds self validating post-install and post-upgrade hooks to the charts.

### Purpose of the change
Fixes #46 

### What the code does
Makes all cluster specs overridable and adds self-validating post-install and post-upgrade hooks to the charts.

### How to verify it
User should be able to follow the instructions mentioned in the documentation to install/delete a bookkeeper operator and bookkeeper cluster using charts via helm v3. User should be able to modify any of the bookkeeper cluster specs through the values.yaml file.

Tested 3 different scenarios for naming the resources
- `fullNameOverride` is mentioned in values.yaml file (resource name uses the value provided for this field)
- `fullNameOverride` is not mentioned, release name same as chart name (resource name is same as the chart name)
- `fullNameOverride` is not mentioned, release name is different from chart name (resource name is in the given format `<release-name>-<chart-name>`)
